### PR TITLE
[NP-5489] Send risk rating ucj data.rank to IdentityMind

### PR DESCRIPTION
**Pull request by Mykola97**
Add predicate to check if data was updated
https://nanopay.atlassian.net/browse/NP-5489

### CODE REVIEW

File: src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction'
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        if ( ! ( obj instanceof X ) ) return false;
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js:27]: When does this happen?

File: src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js:7]: We don't already have a predicate which checks for an update?

Comment (KolombetM) [reply src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js:7]: I found only foam.nanos.ruler.predicate.NewEqOld which checks whole object

File: src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js (hunk 1 above)
Comment (KolombetM) [reply src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js:27]: I don't know, was using another code as an example
Screen Shot 2021-09-29 at 3 20 20 PM

File: src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js (hunk 2 above)
Comment (Nauna) [reply src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js:7]: So despite the above, I have a comment on this.
What we need here, is a way to know when a ucj data is updated on purpose and when a ucj data is just being typed in by user in the wizard, which is always doing updates to save the data.

Generally thats why we have the status of UCJ's .... so in general this should be unnecessary predicate.. however ... the riskRating which is what this is for is going to be a little different cause we grant that immediately.

thinking we add some model level validation to RiskRating, then the rule for IDM can rely on the status of the ucj...if that changes to granted...

File: src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js (hunk 1 above)
Comment (Nauna) [reply src/foam/nanos/crunch/predicate/CapabilityDataUpdated.js:27]: there is no shortage of people who have done this ... me, patrick, mayowa, chanman, eric ... list goes on ... I also just copied.
It was probably just a precaution at some point cause X is cast next.

**Comment by Mykola97** [comment]
Not required after discussion

**Pull request closed without merging**